lib/mix/lib/mix/dep/lock.ex

```diff
@@ -62,4 +63,11 @@ defmodule Mix.Dep.Lock do
   defp lockfile do
     Mix.Project.config[:lockfile]
   end
+
+  defp check_lockfile_conflicts(lockfile, info) do
+    if Regex.run(~r/<<<<<<<|=======|>>>>>>>|\|\|\|\|\|\|\|/, info) do
```

Comment:
I think we can even just do a bunch of `String.contains?` calls tbh, a regex feels like overhead here :\